todoCLI/todoCLI.js

```diff
@@ -0,0 +1,153 @@
+const rl = require(`readline`);
+const fs = require(`fs`);
+
+const menu= `(V) View ✏︎ (n) New ✏︎ (cX) Complete  ✏︎  (dX) Delete ✏︎ (q) Quit\n`;
+
+console.log(`\n\nWelcome to Todo CLI!\n----------------------`);
+const interface = rl.createInterface ( {
+	input: process.stdin,
+	output: process.stdout
+});
+
+waitingForUser();
+
+function waitingForUser () {
+  interface.question(menu, (command) => {
+    let number = parseFloat(command.replace(/[cd]/ , ''))
+    command = command.replace(number, '')
+    switch (command) {
+      case `q`:
+        process.exit();
+        break;
+      case `V`:
+        viewToDoList();
+        break;
+      case `n`:
+        addTask();
+        break;
+      case `c`:
+        completeTask(number);
+        break; 
+      case `d`:
+        deleteTask(number);
+        // console.log(`You typed dX`);
+        break;
+      default:
+        console.log(`Invalid entry\n`);
+        waitingForUser();
+        break;
+      }
+  });
+}
+
+function viewToDoList () {
+  fs.readFile('./toDoList.json', (err, json) => {
+    if (err) { 
+      console.log(`\n   ¯\\\_(ツ)_/¯      You haven't started a list yet. Type 'n' to start one\n`)
+      // waitingForUser();
+    } 
+    else {
+      tasks = JSON.parse(json).tasks;
+      for (let i = 0; i < tasks.length; i++) {
+        const task = tasks[i];
+        if (task.status) { 
+          console.log(`${i+1} [ X ] ${task.description}`);
+        } else {
+          console.log(`${i+1} [   ] ${task.description}`);
+        }
+      }
+      // waitingForUser();
+    }
+    waitingForUser();
+  })
+}
+
+function addTask () {
+  interface.question(`Enter new task: `, (task) => {
+    if (task === "") { 
+      console.log(`Can't add an empty task, please enter task`)
+      waitingForUser()
+    } else {
+      let newTask = {"description": task, status: false}
+      fs.readFile(`./toDoList.json`, (err, json) => {
+        let myToDoList = { "tasks" :[]};
+        if (err) {
+          myToDoList.tasks.push(newTask)
+          writeJSON(myToDoList)
+        } else {
+          myToDoList.tasks = JSON.parse(json).tasks;
+          myToDoList.tasks.push(newTask)
+          writeJSON(myToDoList);
```

Comment:
I think you need to rework the recursive nature (like I said in an earlier comment) so that these functions don't have to call `waitingForUser` at all, but even if you didn't, you want to keep a certain degree of "symmetry". Right now, `addTask` calls `waitingForUser()` explicitly in the "empty task" `if`, and then _indirectly_ through `writeJSON`, which is hard to tell without looking at it's source. Before going there, I was puzzled as to how the successful addition of a task would go back to the menu, since I didn't see the call to `waitingForUser` in that case. 

`writeJSON` should do just that, writing the JSON file, and then after returning, `addTask` should be the one that calls `waitingForUser()`, after which you will then see you can call it _outside_ of the `if` just once for all your branches (and eventually be removed altogether, as that is not really part of `addTask`'s responsibilities)

Reply:
Notice that in both sides of the condition you're doing
```javascript
myToDoList.tasks.push(newTask);
writeJSON(myToDoList);
```
This means you can pull those lines _out_ after the `if`/`else`, since you want to execute them regardless of the outcome of the condition.